packages/tests/web-test-runner.config.js

```diff
@@ -16,33 +18,149 @@ const removeDevModeLogging = {
+  }
+  return [
+    fromRollup(resolveRemap)(resolveRemapConfig),
+    removeDevModeLoggingPlugin,
+    // Detect browsers without modules (e.g. IE11) and transform to SystemJS
+    // (https://modern-web.dev/docs/dev-server/plugins/legacy/).
+    legacyPlugin,
+  ];
 }
 
+const browserPresets = {
+  // Default set of Playwright browsers to test when running locally.
+  default: ["chromium", "firefox", "webkit"],
+
+  // Browsers to test during automated continuous integration.
+  // https://saucelabs.com/platform/supported-browsers-devices
+  // https://wiki.saucelabs.com/display/DOCS/Platform+Configurator#/
+  ci: [
+    "chromium",
+    "firefox",
+    "webkit",
+    // Many browser configurations don't yet work with @web/test-runner-saucelabs.
+    // See https://github.com/modernweb-dev/web/issues/472.
+    "sauce:Windows 10/firefox@68", // Current ESR
+    //"sauce:Windows 10/chrome@latest-3", // Fails with no error message
+    //"sauce:macOS 10.15/safari@latest", // Timeout on "elements with custom properties can move between elements"
+    //"sauce:Windows 10/MicrosoftEdge@18", // Browser start timeout
+    //"sauce:Windows 7/internet explorer@11", // Browser start timeout
+  ],
+};
+
+function getBrowsers() {
+  return (process.env.BROWSERS || "preset:default")
+    .split(",")
+    .map(parseBrowser)
+    .flat();
+}
+
+let sauceLauncher;
+function makeSauceLauncherOnce() {
+  if (!sauceLauncher) {
+    const user = (process.env.SAUCE_USERNAME || "").trim();
+    const key = (process.env.SAUCE_ACCESS_KEY || "").trim();
+    if (!user || !key) {
+      throw new Error(
+        "To test on Sauce, set the SAUCE_USERNAME" +
+          " and SAUCE_ACCESS_KEY environment variables."
+      );
+    }
+    sauceLauncher = createSauceLabsLauncher({ user, key });
+  }
+  return sauceLauncher;
+}
+
+/**
+ * Recognized formats:
+ *
+ *   - "browser"
+ *     Local playwright
+ *     E.g. "chromium", "firefox"
+ *
+ *   - "sauce:os/browser@version"
+ *     Sauce Labs
+ *     E.g. "sauce:macOS 10.15/safari@latest"
+ *
+ *   - "preset:name"
+ *     Expand one of the preset sets of browsers
+ *     E.g. "preset:default", "preset:ci"
+ */
+function parseBrowser(browser) {
+  browser = browser.trim();
+  if (!browser) {
+    return [];
+  }
+
+  if (browser.startsWith("preset:")) {
+    const preset = browser.substring("preset:".length);
+    const entries = browserPresets[preset];
+    if (!entries) {
+      throw new Error(
+        `Unknown preset "${preset}", please pick one of: ` +
+          Object.keys(browserPresets).join(", ")
+      );
+    }
+    return entries.map(parseBrowser).flat();
+  }
+
+  if (browser.startsWith("sauce:")) {
+    // Note this is the syntax used by WCT. Might as well use the same one.
+    const match = browser.match(/^sauce:(.+)\/(.+)@(.+)$/);
+    if (!match) {
+      throw new Error(
+        `Invalid Sauce browser string, ` +
+          `expected format "sauce:os/browser@version", ` +
```

Comment:
maybe add common examples too, since I always get it wrong and have to dig for whether there are spaces, capital letters, etc. At least `sauce:Windows 7/internet explorer@11`

Reply:
Done.